src/autodiff.jl

```diff
@@ -28,17 +30,15 @@ struct StateJacobian <: AbstractJacobian end
 struct ControlJacobian <: AbstractJacobian end
 abstract type AbstractHessian end
 
-function _init_seed!(t1sseeds, coloring, ncolor, nmap)
-    t1sseedvec = zeros(Float64, ncolor)
-    @inbounds for i in 1:nmap
-        for j in 1:ncolor
-            if coloring[i] == j
-                t1sseedvec[j] = 1.0
-            end
+KA.@kernel function _init_seed!(t1sseeds, t1sseedvecs, coloring, ncolor)
+    i = @index(Global, Linear)
+    t1sseedvecs[:,i] .= 0
```

Comment:
oh, I see! `t1seedvecs` is now a 2d array. What size will it take on memory for bigger case? Also, I think we could initiate `t1sseedvecs` to 0.0 before launching the kernel, if each index `i` is affected to a particular row

Reply:
The same than the compressed Jacobian. So ncolors x output dimension.

Reply:
It shouldn't be a problem. Plus after this call it should be gc'ed.